gammapy/irf/effective_area.py

```diff
@@ -246,55 +246,32 @@ def max_area(self):
         cleaned_data = self.data.data[np.where(~np.isnan(self.data.data))]
         return cleaned_data.max()
 
-    def find_energy(self, aeff, reverse=False):
+    def find_energy(self, aeff, emin=None, emax=None):
         """Find energy for given effective area.
 
-        A linear interpolation is performed between the two nodes closest to
-        the desired effective area value. By default, the first match is
-        returned (use `reverse` to search starting from the end of the array)
-
-        TODO: Move to `~gammapy.utils.nddata.NDDataArray`
-
         Parameters
         ----------
         aeff : `~astropy.units.Quantity`
             Effective area value
-        reverse : bool
-            Reverse the direction, i.e. search starting from the end of the array
+        emin : `~astropy.units.Quantity`
+            Lower bracket value in case solution is not unique.
+        emax : `~astropy.units.Quantity`
+            Upper bracket value in case solution is not unique.
 
         Returns
         -------
         energy : `~astropy.units.Quantity`
             Energy corresponding to aeff
         """
-        valid = np.where(self.data.data > aeff)[0]
-        idx = valid[0]
-        if reverse:
-            idx = valid[-1]
-
-        if not reverse:
-            # Return lower edge if first bin is selected
-            if idx == 0:
-                energy = self.energy.lo[idx].value
-            # Perform linear interpolation otherwise
-            else:
-                energy = np.interp(
-                    aeff.value,
-                    (self.data.data[[idx - 1, idx]].value),
-                    (self.energy.nodes[[idx - 1, idx]].value),
-                )
-        else:
-            # Return upper edge if last bin is selected
-            if idx == self.data.data.size - 1:
-                energy = self.energy.hi[idx].value
-            # Perform linear interpolation otherwise
-            else:
-                energy = np.interp(
-                    aeff.value,
-                    (self.data.data[[idx, idx + 1]].value),
-                    (self.energy.nodes[[idx, idx + 1]].value),
-                )
-        return energy * self.energy.unit
+        from ..spectrum.models import TableModel
+
+        if emin is None:
+            emin = self.energy.nodes[0]
+        if emax is None:
+            emax = self.energy.nodes[-1]
+
+        aeff_spectrum = TableModel(self.energy.nodes, self.data.data, values_scale="lin")
```

Comment:
I completely agree, that it's a "misuse" of the `TableModel`, but only in terms of API. Technically it solves the problem and reduces code duplication, which I found preferable in this case. Maybe it could be implemented as a helper function instead...